### PR TITLE
Add the vcs option to the config file

### LINKED ISSUE #37: Optional: remove vcs for a new temporary project
If the user do not want to active vcs by default for new project generated by cargo, he will be able to add this to the config file

### PR DESCRIPTION
**Pull request by yozhgoor**
closes #37

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ pub struct Config {
     pub editor_args: Option<Vec<String>>,
     pub temporary_project_dir: PathBuf,
     pub git_repo_depth: Option<Depth>,
+    pub vcs: Option<bool>,
```
Comment (cecton) [inline src/config.rs:13]: You should take an arbitrary string and pass it to cargo as is if specified

Comment (yozhgoor) [reply src/config.rs:13]: To allow the user to choose another vcs 
Not a bad idea, lemme check this

File: src/run.rs [hunk 2]
```diff
@@ -76,6 +77,18 @@ pub fn generate_tmp_project(
...
+            match arg.as_str() {
+                "Pijul" | "pijul" => command.args(["--vcs", "pijul"]),
+                "Fossil" | "fossil" => command.args(["--vcs", "fossil"]),
+                "HG" | "hg" | "Hg" => command.args(["--vcs", "hg"]),
+                "None" | "non" => command.args(["--vcs", "none"]),
+                _ => command.args(["--vcs", "git"]),
+            };
+        } else {
+            command.args(["--vcs", "git"]);
+        }
```
Comment (cecton) [inline src/run.rs:90]:
```suggestion
        if let Some(arg) = vcs {
            command.args(["--vcs", arg]);
        }
```

File: src/config.rs [hunk 3]
```diff
@@ -32,6 +33,7 @@ impl Config {
             editor_args: None,
             git_repo_depth: None,
             temporary_project_dir,
+            vcs: Some(String::from("git")),
```
Comment (cecton) [inline src/config.rs:36]:
```suggestion
            vcs: None,
```

**Review by cecton — COMMENTED** [review]
That's why I mean by "just pass the argument as is to cargo"

File: README.md [hunk 4]
```diff
@@ -167,6 +167,23 @@ and `editor_args` to provide its arguments. These settings are unset by default.
...
+```toml
+vcs = "pijul"
+```
+
+The possible values are
+
+* Pijul
+* Fossil
+* Hg
+* None
```
Comment (cecton) [inline README.md:183]: It won't work with the capital cases

```suggestion
* pijul
* fossil
* hg
* none
* ... and all other option cargo normally propose
```

File: README.md [hunk 5]
```diff
@@ -167,6 +167,23 @@ and `editor_args` to provide its arguments. These settings are unset by default.
...
+
+If you do not add the option or if the value is `git` or something else that is not in the possible value, cargo-temp will use the default value.
```
Comment (cecton) [inline README.md:185]:
```suggestion
The `--vcs` value will be passed as is to cargo.
```

File: README.md [hunk 6]
```diff
@@ -167,6 +167,23 @@ and `editor_args` to provide its arguments. These settings are unset by default.
     editor_args = [ "--wait", "--new-window" ]
     ```
 
+### Use a VCS
+
+By default, cargo-temp will use Git as the default vcs for you projects, you can change that in the config file with the vcs option.
```
Comment (cecton) [inline README.md:172]:
```suggestion
By default, cargo-temp will use the default cargo VCS for your projects (which is normally git), you can change that in the config file with the `vcs` option.
```

**Review by cecton — COMMENTED** [review]
Note that the default VCS can be configured at cargo's level (in `~/.cargo/config.toml`): https://doc.rust-lang.org/cargo/reference/config.html#cargo-newvcs

I don't personally have a use case for this feature.

**Pull request merged**